robot_sort/robot_sort.py

```diff
@@ -101,21 +101,32 @@ def sort(self):
         self.set_light_on()
 
         while self.light_is_on():
-            #at the beginning of the loop, I need to turn the light off
+            #start of while loop that continues as the light is off
             self.set_light_off()
-
-            #need to created a loop inside of this loop while robot can move right
+            #while the robot can move right
             while self.can_move_right():
-                #in this loop I'll have robot swap items out 
+                #here the robot is swapping the floor item with the held item then moving right
                 self.swap_item()
                 self.move_right()
-                #need a way to compare the value of item held and item on floor
-
-                #need a case here for when item being held is greater than the floor item
-                #little lost on how to do this....feel like I've got a good start
-
-
-        #loop should start with turning light off so loop will run
+                #this condition will compare the two items and if the held item is greater, will return 1
+                if self.compare_item() == 1:
+                    self.swap_item()
+                    self.move_left()
+                    self.swap_item()
+                    self.move_right()
+                    self.set_light_on()
+                #this will run if the held item equal anything lesser than the floor item
+                else:
+                    self.move_left()
+                    self.swap_item()
+                    self.move_right()
+
+            if self.light_is_on():
+                while self.can_move_left():
+                    self.move_left()
+
+                    #runtime of this is probably not the greatest but it does work
```

Comment:
nice! the runtime is fine. it doesn't get to the stretch, but that's how it's supposed to be set up :) great job Joe!